src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/GCPerfSim/GCPerfSimFunctionalCommand.cs

```diff
@@ -0,0 +1,336 @@
+            string gcPerfSimOutputPath = Path.Combine(configuration.output_path, "GCPerfSim");
+            Core.Utilities.TryCreateDirectory(gcPerfSimOutputPath);
+
+            string suitePath = Path.Combine(configuration.output_path, "Suites");
+            string gcPerfSimSuitePath = Path.Combine(suitePath, "GCPerfSim_Functional");
+
+            Core.Utilities.TryCreateDirectory(gcPerfSimSuitePath);
+
+            // For each of the scenarios below:
+            // a. Add the coreruns. 
+            // b. Add the gcperfsim parameters that are pertinent to that run.
+            // c. Add any environment variables that will be related to that run.
+
+            // 1. Normal Server
+            CreateNormalServerSuite(gcPerfSimSuitePath, configuration);
+
+            // 2. Normal Workstation.
+            CreateNormalWorkstationSuite(gcPerfSimSuitePath, configuration);
+
+            // 3. LowMemoryContainer.
+            CreateLowMemoryContainerSuite(gcPerfSimSuitePath, configuration);
+
+            // 4. HighMemoryLoad.
+            CreateHighMemoryLoadSuite(gcPerfSimSuitePath, configuration);
+
+            // III. Execute all the functional tests.
+
+            string[] gcperfsimConfigurationFileNames = Directory.GetFiles(gcPerfSimSuitePath, "*.yaml");
+
+            Dictionary<string, GCPerfSimResults> yamlFileResultMap = new Dictionary<string, GCPerfSimResults>();
+
+            foreach (string gcperfsimConfigurationFileName in gcperfsimConfigurationFileNames)
+            {
+                GCPerfSimConfiguration gcperfsimConfiguration  = 
+                    GCPerfSimConfigurationParser.Parse(gcperfsimConfigurationFileName);
+
+                Stopwatch sw = new();
+                sw.Start();
+
+                AnsiConsole.Write(new Rule($"{gcperfsimConfiguration.Name}"));
+                AnsiConsole.WriteLine();
+
+                // run the test
+                GCPerfSimResults gcperfsimResult = GCPerfSimCommand.RunGCPerfSim(gcperfsimConfiguration, settings.Server);
+                string yamlFileName = Path.GetFileName(gcperfsimConfigurationFileName);
+                yamlFileResultMap[yamlFileName] = gcperfsimResult;
+
+                
+                sw.Stop();
+                AnsiConsole.WriteLine($"Time to execute Msec: {sw.ElapsedMilliseconds}");
+            }
+            // Look at the RunCommand from 99 onwards to see how these are generated.
+
+            // IV. Based on the results of the functional tests, we'd want to generate a report.
+            // Looking at all the runs data, we'd want to aggregate and create a markdown table of the following type:
+            // Output should live in Results.md in the output folder.
+            // TODO: If the run fails, what are the commands, failures and any debugging information that'll be helpful.
+            // | Yaml | Scenario | Pass / Fail  |
+            // | ---- | -------- | ------------ |
+            // | Normal_Server.yaml | 0gb | Pass |
```

Comment:
NIT: Make use of the following rather than pass / fail. 

Tick (Check Mark): ✅
Cross (Incorrect Mark): ❌